File: Lab_02/src/main.py
import random
import time


def iteration(x):
    return x ** 2 - x ** 2 + x * 4 - x * 5 + x + x


def run_experiment(n, x):
    start_time = time.time()
    for _ in range(n):
        iteration(x)

    sum_time = time.time() - start_time
    print(f"Время {n} итераций: {sum_time:.6f}")


def is_number(s):
    try:
        int(s)
        return True
    except ValueError:
        return False


def main():
    while True:
        user_input = input("Введите количесто итераций (или не число для выхода из программы)")

        if not is_number(user_input):
            print("Работа программы завершена")
            break

        n = int(user_input)
        x = random.randint(1, 100000)
        run_experiment(n, x)


main()

File: Lab_02/src/main.cpp
#include <iostream>
#include <chrono>
#include <string>
#include <cctype>
#include <cstdlib>

double iteration(double x) {
    return x*x - x*x + x*4 - x*5 + x + x;
}

void run_experiment(int n, int x) {
    auto start_time = std::chrono::high_resolution_clock::now();

    for (int i = 0; i < n; ++i) {
        iteration(x);
    }

    auto end_time = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> sum_time = end_time - start_time;

    std::cout << "Time of " << n << "iterations: " << sum_time.count() << std::endl;
}

bool is_number(std::string s) {

    for (int i = 0; i < s.length(); ++i) {
        if (!isdigit(s[i])) {
            return false;
        }
    }
    return true;
}

int main() {
    while (true) {
        std::cout << "Enter number of iterations (or non-number to exit): ";
        std::string input;
        std::cin >> input;

        if (!is_number(input)) {
            std::cout << "Program completed" << std::endl;
            break;
        }

        int n = std::stoi(input);

        int x = rand();
        run_experiment(n, x);
    }

    return 0;
}
